lib/matplotlib/collections.py

```diff
@@ -2195,11 +2195,45 @@ def draw(self, renderer):
         renderer.close_group(self.__class__.__name__)
         self.stale = False
 
+    def contains(self, event):
+        # docstring inherited
```

Comment:
is there a way to fast-path the often-used pcolormesh case of a rectilinear grid (lat and lon are each 1-d)?

Reply:
Sort-of. I'm not sure it is doable in QuadMesh currently because it is so flexible with the mesh coordinates, so somehow we would need to store some information about what x/y grid was passed in. That case is already handled in PcolorImage though:
https://github.com/matplotlib/matplotlib/blob/cb8680f589c75fb666fdae1ab8194ead1eca468b/lib/matplotlib/image.py#L1318-L1329
(also note that I still get quite good performance for 400x400 size meshes which is already pretty small boxes for investigating coordinate output)

Thinking about this more generally, I wonder if we could consolidate some of these pcolor-style plots into a more generic Collection that would defer choosing which one of PcolorImage/QuadMesh/PolyCollection to draw until the very end, but leaving all of the options available. For instance, if someone wants `shading=gouraud` we would choose QuadMesh for the rendering. I think this was sort of the idea behind `pcolorfast()` (?), but that will return _different_ collections depending on what the input is, which may be somewhat confusing for modifying properties later?

Reply:
I think in general unifying the three pcolor methods would be great - I am personally not a fan of different ones for different purposes; in general if there is a performance issue, one just rasterizes anyhow and performance is usually more than adequate.